### PR TITLE
fix access to identifiers that are reserved keywords

### LINKED ISSUE #61981: Can't access some identifiers on a Dictionary
### Godot version

803d25a

### System information

Windows 10 x64

### Issue description

Doing e.g.
```GDScript
var dict = {}
dict.class = "Node"
```
results in `Expected identifier after "." for attribute access.` and `Expected end of statement after expression, found "class" instead.`, spammed tens of times. The identifier is incorrectly recognized as a keyword.

Works correctly in 3.x

### Steps to reproduce

1. Use code above

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by ajreckof**
Fixes #61981 

All reserved keywords listed here were interpreted as tokens and returned false when tested with `is_identifier()`. So i made it possible to indicate that a token can be interpreted as an identifier whatever is type. Then in `parse_attribute` when we know we want exclusively identifiers i marked keywords as identifiers.

### CODE REVIEW

File: modules/gdscript/gdscript_tokenizer.h [hunk 1]
```diff
@@ -169,6 +169,7 @@ class GDScriptTokenizer {
 		int cursor_position = -1;
 		CursorPlace cursor_place = CURSOR_NONE;
 		String source;
+		bool marked_as_indentifier;
```
Comment (akien-mga) [inline modules/gdscript/gdscript_tokenizer.h:172]: Typo + default value:
```suggestion
		bool marked_as_identifier = false;
```

Comment (ajreckof) [reply modules/gdscript/gdscript_tokenizer.h]: thanks for the typo correction and the advice i corrected the typo at all the places as i was consistent with my typo thanks to autocompletion

**Comment by ajreckof** [comment]
I just realised that instead of creating a variable to mark them as identifier i could directly change the type of the token as the function that gets the identifier from the token isn't using the type but another variable which store the string directly. 
Changing the type will make clearer code as it will make only changes in the `get_attribute` function

**Comment by anvilfolk** [comment]
I've never fooled around with the tokenizer so I don't feel confident reviewing :)

Couple of thoughts though!

1. Is there a chance you could add a couple of tests that showcase this working and so the same error doesn't creep back up later?
2. Might there be a situation where something that should be a reserved name can now become just an identifier and cause a bug? Or is that just not possible?

**Comment by ajreckof** [comment]
> 1. Is there a chance you could add a couple of tests that showcase this working and so the same error doesn't creep back up later?

That seems like a super good idea I'll try to make some but i don't know how to make one so I'll have to look into it.

> 2. Might there be a situation where something that should be a reserved name can now become just an identifier and cause a bug? Or is that just not possible?

If it reaches this point and it is not interpreted as an identifier it will just raise an error at the next line. That's why we can here safely modify it.

**Comment by anvilfolk** [comment]
I tried looking through the source code some more to try to give this a review but it's a little too much for me atm :)

The added test looks good though! From now we can forever make sure that you can use reserved keywords as dictionary keys in Lua-style accessors!

**Review by anvilfolk — APPROVED** [review]
Actually, I ran this in debug mode and I do believe that it makes sense! :)

The function changed is already specifically for attribute lookups. The changes make sure that if what comes after the `.` is basically text (the call to `is_node_name()`), then it should be interpreted as an identifier. Again, _only_ specifically within the context of an attribute lookup, i.e. `base.attribute`.

Anything else that isn't text, when `is_node_name()` returns false, will still raise an error due to the `consume` function wanting to see an identifier there.

**Review by adamscott — APPROVED** [review]
Approved by the GDScript team on a PR review meeting.

Thanks for your contribution!

(as discussed, the PR will be ready when more cases will be added to the test)

**Comment by YuriSizov** [comment]
> (as discussed, the PR will be ready when more cases will be added to the test)

Marked as a draft until then. Feel free to mark as ready when appropriate.

**Comment by adamscott** [comment]
It's ready to review! Hurray! :tada:

**Review by YuriSizov — APPROVED** [review]

**Comment by YuriSizov** [comment]
Thanks!

**Pull request merged**